nvflare/private/fed/client/client_aux_runner.py

```diff
@@ -104,23 +105,27 @@ def _send_fnf_requests(self):
         sleep_time = 0.5
         while True:
             time.sleep(sleep_time)
-            if self.asked_to_stop or self.abort_signal.triggered:
+            if self.abort_signal.triggered:
                 break
 
-            with self.fnf_lock:
-                if len(self.fnf_requests) <= 0:
+            if len(self.fnf_requests) <= 0:
```

Comment:
Just to confirm this len function does not need to be inside the lock?

Reply:
The with self.fnf_lock statement goes down a few lines to reduce locking time.  len is read-only so we can move it out of the self.fnf_lock.